src/jdk.jlink/share/classes/jdk/tools/jlink/builder/DefaultImageBuilder.java

```diff
@@ -144,16 +143,34 @@ public Platform getTargetPlatform() {
     private final Map<String, String> launchers;
     private final Path mdir;
     private final Set<String> modules = new HashSet<>();
-    private Platform platform;
+    private final Platform platform;
 
     /**
      * Default image builder constructor.
      *
      * @param root The image root directory.
+     * @param launchers mapping of launcher command name to their module/main class
      * @throws IOException
+     * @throws NullPointerException If any of the params is null
      */
     public DefaultImageBuilder(Path root, Map<String, String> launchers) throws IOException {
+        this(root, launchers, Platform.UNKNOWN);
+    }
+
+    /**
+     * Default image builder constructor.
+     *
+     * @param root The image root directory.
+     * @param launchers mapping of launcher command name to their module/main class
+     * @param targetPlatform target platform of the image
+     * @throws IOException
+     * @throws NullPointerException If any of the params is null
+     * @since 21
```

Comment:
you can drop this as this is not a public API.

Reply:
Removed in latest update of the PR.